cmd/dep/init.go

```diff
@@ -210,6 +211,43 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// getProjectPropertiesFromVersion takes a gps.Version and returns a proper
+// gps.ProjectProperties with Constraint value based on the version type.
+func getProjectPropertiesFromVersion(v gps.Version) gps.ProjectProperties {
+	pp := gps.ProjectProperties{}
+	switch v.Type() {
```

Comment:
So, we actually need to do a little more work here up front. It's not a guarantee of the interface (yet - this may change), but gps has a concept of "[version pairing](https://godoc.org/github.com/sdboyer/gps#PairedVersion)" - basically, combining a revision with a branch or tag. This information is not encoded in the `gps.Version.Type()` method, so we need to tease it out first, because - per the spec - we _never_ want to put revisions in Gopkg.toml, if we can avoid it.

There's a rough skeleton to follow here in [`status.go`](https://github.com/golang/dep/blob/e0256827b951cb857673fae69f34ee71ccc043ea/cmd/dep/status.go#L303-L324).